custom_components/philips_airpurifier_http/fan.py

```diff
@@ -118,8 +119,15 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     client = await hass.async_add_executor_job(
         lambda: HTTPAirClient(config[CONF_HOST], False)
     )
+    unique_id = None
 
-    device = PhilipsAirPurifierFan(hass, client, name)
+    wifi = await hass.async_add_executor_job(client.get_wifi)
+    if PHILIPS_MAC_ADDRESS in wifi:
+        unique_id = wifi[PHILIPS_MAC_ADDRESS]
+    else:
+        unique_id = config[CONF_UNIQUE_ID]
```

Comment:
I see a couple of potential issues with this:

1. In a future where there is a config flow setup, there will have to be a Unique ID field. However, if we detect the mac address, we will ignore the unique_id the user entered, which would be confusing
2. `CONF_UNIQUE_ID` seems to be intended to be used for template / generic entities, not custom components
3. We still would have to validate that this is unique.
4. What if the user doesn't enter a unique_id and adds 2 devices that we can't get Mac addresses for?

I'm really not convinced about `unique_id` until we know there's a reliable way to obtain one from every device. Maybe I need to read up more on Home Assistant best practices :) But let's revert that and leave it out for now.

Reply:
Ping @ronaldt80, do you mind reverting the `unique_id` changes, so I can merge the `_http` rename and we can go from there?